### PR TITLE
24 team statistics rival

### LINKED ISSUE #24: Team Statistics - rival
Returns string: Name of the opponent that has the highest win percentage against the given team.

### PR DESCRIPTION
**Pull request by knestler**
fix: #24 feature complete - also added to gitignore to quit tracked changes.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,2 +1,3 @@
 coverage
 .vscode/
+.DS_Store
```
Comment (stephenfabian) [inline .gitignore]: Is there potential issue with ds_store getting added to git ignore?

**Review by stephenfabian — DISMISSED** [review]
Thanks for explaining .merge

File: lib/stat_tracker.rb [hunk 2]
```diff
@@ -502,10 +524,55 @@ def favorite_opponent #issue # 29 - Fail due to not written
...
+        if row[:away_goals] < row[:home_goals]
+          rivals_wins << row[:home_team_id]
+        end
+      end
+      if row[:home_team_id] == team_id.to_i
+        if row[:home_goals] < row[:away_goals]
+          rivals_wins << row[:away_team_id]
+        end
+      end
+      # require "pry"; binding.pry
```
Comment (aleish-m) [inline lib/stat_tracker.rb]: Don't forget to remove your pry 😉

File: lib/stat_tracker.rb [hunk 3]
```diff
@@ -502,10 +524,55 @@ def favorite_opponent #issue # 29 - Fail due to not written
...
+        end
+      end
+      if row[:home_team_id] == team_id.to_i
+        if row[:home_goals] < row[:away_goals]
+          rivals_wins << row[:away_team_id]
+        end
+      end
+      # require "pry"; binding.pry 
+    end
+      rivals_wins_hash = rivals_wins.tally
```
Comment (aleish-m) [inline lib/stat_tracker.rb:541]: Interesting, I had not heard of the tally method. It's really cool! Thanks for introducing me to it.

**Review by aleish-m — DISMISSED** [review]
Looks good. I'll approve it and the pry can get removed later.

**Review by aleish-m — APPROVED** [review]

**Review by stephenfabian — APPROVED** [review]
thanks for removing pry

File: lib/stat_tracker.rb (hunk 2 above)
Comment (knestler) [reply lib/stat_tracker.rb]: removed! thanks!

**Pull request merged**